api/workflow_delete.go

```diff
@@ -0,0 +1,35 @@
+package api
+
+import (
+	"github.com/mesg-foundation/core/systemservices/workflow"
+)
+
+// DeleteWorkflow stops and deletes a workflow. id can be unique id or name.
+func (a *API) DeleteWorkflow(id string) (err error) {
+	return newWorkflowDeletor(a).Delete(id)
+}
+
+// workflowDeletor provides functionalities to stop and delete a workflow.
+type workflowDeletor struct {
+	api *API
+}
+
+// newWorkflowCreator returns a new workflow creator with given api.
+func newWorkflowDeletor(api *API) *workflowDeletor {
+	return &workflowDeletor{
+		api: api,
+	}
+}
+
+// Delete stops and deletes a workflow. id can be unique id or name.
+func (w *workflowDeletor) Delete(id string) (err error) {
```

Comment:
Same here - move Delete body to DeleteWorkflow

Reply:
I kind of agree with this, but we follow the same pattern all around the _api_. I think, for now we should keep this one as it's and if we want to make a change about this pattern we can do it for all methods together in a different PR.